phocoa/framework/WFAuthorization.php

```diff
@@ -535,11 +535,15 @@ function login($username, $password, $passIsToken = false)
      *
      * @param object WFAuthorizationInfo The new authorization info to set for the current session.
      */
-    function loginAsAuthorizationInfo($authInfo, $authorizeRecentLogin = true)
+    function loginAsAuthorizationInfo($authInfo, $authorizeRecentLogin = true, $temporary = false)
     {
         if (!$authInfo instanceof WFAuthorizationInfo) throw new WFException("WFAuthorizationInfo or subclass required.");
 
         $this->authorizationInfo = $authInfo;
+        
+        if($temporary){
+      			return;
+      		}
```

Comment:
in my implementation, i just fetch the user out of the db based on the request header and set this for the request only. by adding "temporary" it prevents writing into the session and influencing other requests however everything that is using the shared manager static method can see this "temporary" user until the request is over.